test/setup/loadEnvFiles.ts

```diff
@@ -0,0 +1,36 @@
+import dotenv from 'dotenv';
+import dotenvExpand from 'dotenv-expand';
+
+function loadEnvFile(
+	envFilePath: string,
+	providedEnvvars: Record<string, string> = {}
+): Record<string, string> {
+	const result = dotenv.config( {
+		override: true,
+		path: envFilePath,
+		processEnv: providedEnvvars
+	} );
+
+	if ( result.error ) {
+		throw result.error;
+	}
+
+	const envVarsResult = dotenvExpand.expand( {
+		parsed: providedEnvvars, ignoreProcessEnv: true
+	} );
```

Comment:
Framing is confusing - it looks like you're returning the variables from the config file into `results`, and then failing to include that in `dotenvExpand.expand`. Would just fetching `{ error }` instead of `result` work?

Reply:
Naming the variable `providedEnvvars` is part of the issue, I think - the variable name does not suggest it to be mutable.

Reply:
Certainly you have it right that `provided*` var naming for me is always chosen to indicate immutability. So will change that and see what can be done to clarify things here more.

The two libraries I'm using here work fairly implicitly and their APIs kinda suck, so I blame that a bit.